test/fw/ptl/utils/pbs_dshutils.py

```diff
@@ -1170,26 +1172,40 @@ def run_copy(self, hosts=None, src=None, dest=None, sudo=False, uid=None,
 
             # Remote copy if target host is remote or if source file/dir is
             # remote.
-            if ((not islocal) or (':' in src)):
+            if srchost:
+                srchost = socket.getfqdn(srchost)
+            if ((not islocal) or (srchost)):
                 copy_cmd = copy.deepcopy(self.copy_cmd)
-                if not preserve_permission:
-                    copy_cmd.remove('-p')
-                if copy_cmd[0][0] != '/':
-                    copy_cmd[0] = self.which(targethost, copy_cmd[0],
-                                             level=level)
-                cmd += copy_cmd
-                if recursive:
-                    cmd += ['-r']
-                if runas and runas.port:
-                    cmd += ['-P', runas.port]
-                cmd += [src]
-                if islocal:
-                    cmd += [dest]
+                targethost = socket.getfqdn(targethost)
+                if (srchost == targethost):
```

Comment:
Is this really necessary?  If you don't use 'host:' as part of the paths, scp will do a local copy.

Reply:
Yes but the local copy will be done on the local machine but not on the remote machine. These changes are done to a cp on a remote machine.
Incase we want to copy a file from path a to path b on a mom machine or a client machine. This will be helpful.

Reply:
Ahh, got it.  I see the use case now.